extensions/guacamole-vault/modules/guacamole-vault-base/src/main/java/org/apache/guacamole/vault/util/FieldParsingUtil.java

```diff
@@ -0,0 +1,163 @@
+         *
+         * @param username
+         *     The username portion of the field value from the vault.
+         *
+         * @param domain
+         *     The domain portion of the field value from the vault.
+         */
+        private DomainAndUsername(
+                @NonNull String username, @NonNull String domain) {
+            this.username = username;
+            this.domain = domain;
+        }
+
+        /**
+         * Return the value of the username as extracted from the vault field.
+         * If the domain is null, this will be the entire field value.
+         *
+         * @return
+         *     The username value as extracted from the vault field.
+         */
+        public String getUsername() {
+            return username;
+        }
+
+        /**
+         * Return the value of the domain as extracted from the vault field.
+         * If this is null, it means that no domain was found in the vault field.
+         *
+         * @return
+         *     The domain value as extracted from the vault field.
+         */
+        public String getDomain() {
+            return domain;
+        }
+
+        /**
+         * Return true if a domain was found in the vault field, false otherwise.
+         *
+         * @return
+         */
+        public boolean hasDomain() {
+            return this.domain != null;
+        }
+
+    }
+
+    /**
+     * Strip off a Windows domain from the provided username, if one is
+     * present. For example: "DOMAIN\\user" or "user@DOMAIN" will both
+     * be stripped to just "user". Note: neither the '@' or '\\' characters
+     * are valid in Windows usernames.
+     *
+     * @param vaultField
+     *     The raw field value as retrieved from the vault. This might contain
+     *     a Windows domain.
+     *
+     * @return
+     *     The provided username with the Windows domain stripped off, if one
+     *     is present.
+     */
+    public static DomainAndUsername splitWindowsUsernameFromDomain(String vaultField) {
+
+        // If it's the down-level logon format, return the extracted username and domain
+        Matcher downLevelLogonMatcher = DOWN_LEVEL_LOGON_NAME_PATTERN.matcher(vaultField);
+        if (downLevelLogonMatcher.matches())
+            return new DomainAndUsername(
+                    downLevelLogonMatcher.group("username"),
+                    downLevelLogonMatcher.group("domain"));
+
+        // If it's the user principal format, return the extracted username and domain
+        Matcher userPrincipalMatcher = USER_PRINCIPAL_NAME_PATTERN.matcher(vaultField);
+        if (userPrincipalMatcher.matches())
+            return new DomainAndUsername(
+                    userPrincipalMatcher.group("username"),
+                    userPrincipalMatcher.group("domain"));
+
+        // If none of the expected formats matched, return the username with do domain
+        return new DomainAndUsername(vaultField);
+
+    }
```

Comment:
Rather than establish a general parsing utility class for all future parsing needs, I suggest instead just defining `DomainAndUsername` and let it do-one-thing-and-do-it-well. ie:

`DomainAndUsername.split(someFieldValue)`

rather than:

`FieldParsingUtil.DomainAndUsername` and `FieldParsingUtil.splitWindowsUsernameFromDomain(someFieldValue)`.

Reply:
Sure

Reply:
It's done.